diff -urN boost-numeric-bindings_orig/Makefile.in boost-numeric-bindings/Makefile.in
--- boost-numeric-bindings_orig/Makefile.in	2008-08-16 02:22:29.000000000 -0700
+++ boost-numeric-bindings/Makefile.in	2012-07-24 08:58:53.000000000 -0700
@@ -3,6 +3,8 @@
 
 install:
 	for header in `find boost -name '*.hpp' -o -name '*.h'`; do \
-	  install -D $$header ${PREFIX}/include/boost-numeric-bindings/$$header ; \
-	  done
-	  
+		dir=`dirname $$header`; \
+		mkdir -p ${PREFIX}/include/boost-numeric-bindings/$$dir; \
+		install $$header ${PREFIX}/include/boost-numeric-bindings/$$header; \
+	done
+
