diff -urN healpy.orig/hpbeta/config/config.HPCP healpy/hpbeta/config/config.HPCP
--- healpy.orig/hpbeta/config/config.HPCP	1969-12-31 16:00:00.000000000 -0800
+++ healpy/hpbeta/config/config.HPCP	2012-10-23 09:06:27.000000000 -0700
@@ -0,0 +1,16 @@
+
+EXTERNAL_CFITSIO = yes
+CFITSIO_EXT_LIB = $(CFITSIO_EXT_PREFIX)/lib/libcfitsio.a
+CFITSIO_EXT_INC = $(CFITSIO_EXT_PREFIX)/include
+
+CL = $(CC)
+CCFLAGS_NO_C= $(CFLAGS) $(OMPFLAGS)
+CCFLAGS= $(CCFLAGS_NO_C) -c
+CLFLAGS= -L. -L$(LIBDIR) $(OMPFLAGS) $(LDFLAGS) $(LIBS) -lm
+
+CXXL = $(CXX)
+CXXCFLAGS_NO_C = $(CXXFLAGS) $(OMPFLAGS)
+CXXCFLAGS = $(CXXCFLAGS_NO_C) -c
+CXXLFLAGS = -L. -L$(LIBDIR) $(OMPFLAGS) $(LDFLAGS) $(LIBS)
+
+ARCREATE= ar cr
diff -urN healpy.orig/setup.py healpy/setup.py
--- healpy.orig/setup.py	2012-07-23 17:00:47.000000000 -0700
+++ healpy/setup.py	2012-10-23 09:05:57.000000000 -0700
@@ -10,23 +10,16 @@
     """
     import subprocess
     import re
-
-    try:
-        cc_output = subprocess.check_output(['cc', '--version'],
-                                            stderr = subprocess.STDOUT)
-    except:
-        return False
-
-    return re.search('clang', cc_output) is not None
+    return False
 
 TARGET_DICT = {
-    'linux': 'healpy',
-    'darwin': 'healpy_osx'
+    'linux': 'HPCP',
+    'darwin': 'HPCP'
 }
 
 FLAGS_DICT = {
-    'openmp' : '-fopenmp',
-    'native' : '-march=native'
+    'openmp' : '',
+    'native' : ''
 }
 
 DEFAULT_OPT_DICT = {
