diff -urN cfitsio.orig/configure cfitsio/configure
--- cfitsio.orig/configure	2012-04-13 12:16:16.000000000 -0700
+++ cfitsio/configure	2013-01-07 23:18:23.000000000 -0800
@@ -3892,11 +3892,11 @@
 
     case $system in
         Darwin-[56789]*)
-            SHLIB_LD="$CC -dynamiclib"
+            SHLIB_LD="$CC -dynamiclib -install_name \$(CFITSIO_LIB)/libcfitsio\$(SHLIB_SUFFIX)"
             ;;
         *)
             # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
-            SHLIB_LD="$CC -dynamiclib $C_UNIV_SWITCH"
+            SHLIB_LD="$CC -dynamiclib -install_name \$(CFITSIO_LIB)/libcfitsio\$(SHLIB_SUFFIX)"
             ;;
     esac
 
diff -urN cfitsio.orig/configure.in cfitsio/configure.in
--- cfitsio.orig/configure.in	2012-04-13 12:16:25.000000000 -0700
+++ cfitsio/configure.in	2013-01-07 23:17:47.000000000 -0800
@@ -327,11 +327,11 @@
     changequote(,)
     case $system in
         Darwin-[56789]*)
-            SHLIB_LD="$CC -dynamiclib"
+            SHLIB_LD="$CC -dynamiclib -install_name \$(CFITSIO_LIB)/libcfitsio\$(SHLIB_SUFFIX)"
             ;;
         *)
             # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
-            SHLIB_LD="$CC -dynamiclib $C_UNIV_SWITCH" 
+            SHLIB_LD="$CC -dynamiclib -install_name \$(CFITSIO_LIB)/libcfitsio\$(SHLIB_SUFFIX)"
             ;;
     esac
     changequote([,])
