diff -urN tcl8.5.12.orig/unix/Makefile.in tcl8.5.12/unix/Makefile.in
--- tcl8.5.12.orig/unix/Makefile.in	2012-07-25 09:45:05.000000000 -0700
+++ tcl8.5.12/unix/Makefile.in	2012-08-22 11:18:56.000000000 -0700
@@ -56,7 +56,7 @@
 INCLUDE_INSTALL_DIR	= $(INSTALL_ROOT)$(includedir)
 
 # Path to the private tcl header dir:
-PRIVATE_INCLUDE_DIR	= @PRIVATE_INCLUDE_DIR@
+PRIVATE_INCLUDE_DIR	= $(includedir)
 
 # Directory in which to (optionally) install the private tcl headers:
 PRIVATE_INCLUDE_INSTALL_DIR = $(INSTALL_ROOT)$(PRIVATE_INCLUDE_DIR)
@@ -80,7 +80,7 @@
 HTML_INSTALL_DIR	= $(INSTALL_ROOT)$(HTML_DIR)
 
 # Directory in which to install the configuration file tclConfig.sh
-CONFIG_INSTALL_DIR	= $(INSTALL_ROOT)$(libdir)
+CONFIG_INSTALL_DIR	= $(INSTALL_ROOT)$(bindir)
 
 # Package search path.
 TCL_PACKAGE_PATH	= @TCL_PACKAGE_PATH@
@@ -727,6 +727,7 @@
 	@$(INSTALL_PROGRAM) ${TCL_EXE} "$(BIN_INSTALL_DIR)"/tclsh$(VERSION)@EXEEXT@
 	@echo "Installing tclConfig.sh to $(CONFIG_INSTALL_DIR)/"
 	@$(INSTALL_DATA) tclConfig.sh "$(CONFIG_INSTALL_DIR)"/tclConfig.sh
+	@chmod +x "$(CONFIG_INSTALL_DIR)"/tclConfig.sh
 	@if test "$(STUB_LIB_FILE)" != "" ; then \
 	    echo "Installing $(STUB_LIB_FILE) to $(LIB_INSTALL_DIR)/"; \
 	    @INSTALL_STUB_LIB@ ; \
